src/lib/universal-client.js

```diff
@@ -0,0 +1,638 @@
+/* cometvisu.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+ */
+
+/**
+ * The JavaScript library that implements the CometVisu protocol.
+ * 
+ * @module CometVisu Client
+ * @title CometVisu Client
+ * @reqires jQuery
+ */
+define(
+    [],
+    function() {
+      "use strict";
+
+      /**
+       * Class that handles the communicaton of the client *
+       * 
+       * @class CometVisu
+       * @constructor foo
+       * @param backend {String} name of the backend (cgi-bin|default|oh|openhab|oh2|openhab2)
+       * @param initPath {String|null} optional path to login ressource
+       */
+      function CometVisu(backend, initPath) {
+        this.initPath = initPath;
+
+        // used for backwards compability
+        this.aliases = {
+          'cgi-bin' : 'default',
+          'oh' : 'openhab',
+          'oh2' : 'openhab2'
+        };
```

Comment:
Geschmacksfrage: Ich bevorzuge es, wenn in einem Objekt/Hash die Werte für die Keys linksbündig sind, in diesem Fall wäre dass dann:

    this.aliases = {
      'cgi-bin' : 'default',
      'oh'      : 'openhab',
      'oh2'     : 'openhab2'
    };

Bei diesem kleinen Objekt ist das nicht wirklich relevant, aber weiter unten (z.B. wo die Ajax-Requests zusammen gebaut werden) schaut's für mir unordentlich und schwer zu lesen aus